### PR TITLE
Help: Migrate support history to `react-query`

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

Currently, the support history (active support tickets) selector is used only in one place. It's the only consumer of the `help.supportHistory` state subtree. That makes this part of the state a good candidate for a `react-query` migration.

This PR creates a new `useActiveSupportTicketsQuery` query with a `withActiveSupportTickets` HoC. It updates `HelpContact` to use that query instead of `<QuerySupportHistory />` and `getActiveSupportTickets()`. It also removes a `<QuerySupportHistory />` usage that was not necessary. Finally, it cleans up the query component and the support history state that is now unused.

Note that at this time, this [query was disabled](https://github.com/Automattic/wp-calypso/blob/cc485bfb786b8655f91d4bdde8d6289e848f0ef5/client/state/help/actions.js#L9) temporarily, so we're carrying that over to the `react-query` implementation (see the `ENABLE_SUPPORT_HISTORY` constant). 

#### Testing instructions

* Checkout the branch locally.
* Go to `/help/contact`
* Create a testing support ticket.
* Change `ENABLE_SUPPORT_HISTORY` to `true`.
* Go to `/help/contact` again.
* Verify that the request to `/support-history` works well, and you can see the notice:

### CODE REVIEW

**Comment by jsnajdr** [comment]
I did something similar yesterday in #58988, but didn't get as far as to migrate to React Query 🙂 But let it as draft because @klimeryk has a PR to re-enable the support history again: #57870. There's going to be a conflict with that.

Your PR is better, but there is one thing you should copy: we only care about the _count_ of the active tickets, and you might want to update the query to return just the count. Also, in the `HelpContact` component, I extracted sending the `calypso_help_contact_submit_with_active_tickets` analytics event to a separate method, used four times.

**Comment by tyxla** [comment]
> 

Great feedback, thanks! I've implemented both suggestions (the count and the method). Ready for another look!

File: client/data/help/use-active-support-tickets-query.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export const useActiveSupportTicketsQuery = ( email, queryOptions = {} ) =>
+	useQuery(
+		[ 'activeSupportTickets', email ],
+		() => wpcom.req.get( '/support-history', { email, apiNamespace: 'wpcom/v2' } ),
+		{
+			enabled: !! email && ENABLE_SUPPORT_HISTORY,
+			select: ( { data } ) =>
+				data.filter(
+					( item ) => item.type === 'Zendesk_History' && ACTIVE_STATUSES.includes( item.status )
+				),
```
Comment (jsnajdr) [inline client/data/help/use-active-support-tickets-query.js:19]: One downside of using the `select` function, and selecting only `.length` from it in the HOC, is that the _entire_ unfiltered array is stored in persistent storage:
Screenshot 2021-12-12 at 10 38 31
To store only the count, the query function itself would have to do the filtering:
```js
() => wpcom.req.get().then(({data}) => data.filter().length
```
Or maybe disable persistence for this query completely? Not sure if it makes sense here.

File: client/blocks/inline-help/inline-help-query-support-types.jsx [hunk 2]
```diff
@@ -53,7 +51,6 @@ class QueryInlineHelpSupportTypes extends Component {
 		return (
 			<Fragment>
 				<QueryTicketSupportConfiguration />
-				<QuerySupportHistory email={ this.props.currentUserEmail } />
```
Comment (jsnajdr) [inline client/blocks/inline-help/inline-help-query-support-types.jsx:56]: One benefit of removing `QuerySupportHistory` from here is that the history is not loaded right when you open the help popup, but only after you select the "Contact us" tab.

As a followup, the entire `QueryInlineHelpSupportTypes` should be moved from the `InlineHelpPopover` component down to the `InlineHelpContactView` component which is the actual one that really needs to fetch all these data.

**Review by jsnajdr — APPROVED** [review]

File: client/data/help/use-active-support-tickets-query.js (hunk 1 above)
Comment (tyxla) [reply client/data/help/use-active-support-tickets-query.js:19]: I think it makes sense to remove persistence here TBH. The support ticket can be closed at any time and having the number of open tickets as fresh as possible makes sense. I'll update that before merging.

Comment (tyxla) [reply client/data/help/use-active-support-tickets-query.js:19]: Removed persistence for that query in 42cede6987fb147dba98514eea2e4f8552452da5

File: client/blocks/inline-help/inline-help-query-support-types.jsx (hunk 2 above)
Comment (tyxla) [reply client/blocks/inline-help/inline-help-query-support-types.jsx:56]: Good call. I'll follow up with this in another PR.

Comment (tyxla) [reply client/blocks/inline-help/inline-help-query-support-types.jsx:56]: See #59116.

**Pull request merged**